pkg/destroy/openstack/openstack.go

```diff
@@ -449,31 +457,112 @@ func updateFips(allFIPs []floatingips.FloatingIP, opts *clientconfig.ClientOpts,
 	return nil
 }
 
-func deleteRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
-	logger.Debug("Deleting openstack routers")
-	defer logger.Debugf("Exiting deleting openstack routers")
-
+// deletePortFIPs looks up FIPs associated to the port and attempts to delete them
+func deletePortFIPs(portID string, opts *clientconfig.ClientOpts, logger logrus.FieldLogger) error {
 	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return err
+	}
+
+	fipPages, err := floatingips.List(conn, floatingips.ListOpts{PortID: portID}).AllPages()
+
 	if err != nil {
 		logger.Error(err)
-		return false, nil
+		return err
+	}
+
+	fips, err := floatingips.ExtractFloatingIPs(fipPages)
+	if err != nil {
+		logger.Error(err)
+		return err
+	}
+
+	for _, fip := range fips {
+		logger.Debugf("Deleting FIP %q", fip.ID)
+		err = floatingips.Delete(conn, fip.ID).ExtractErr()
+		if err != nil {
+			// Ignore the error if the FIP cannot be found
+			var gerr gophercloud.ErrDefault404
+			if !errors.As(err, &gerr) {
+				logger.Errorf("Deleting FIP %q failed: %v", fip.ID, err)
+				return err
+			}
+			logger.Debugf("Cannot find FIP %q. It's probably already been deleted.", fip.ID)
+		}
+	}
+	return nil
+}
+
+func getRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) ([]routers.Router, error) {
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return nil, err
 	}
 	tags := filterTags(filter)
 	listOpts := routers.ListOpts{
 		TagsAny: strings.Join(tags, ","),
 	}
 
 	allPages, err := routers.List(conn, listOpts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+
+	allRouters, err := routers.ExtractRouters(allPages)
+	if err != nil {
+		return nil, err
+	}
+	return allRouters, nil
+}
+
+func clearRoutersInferfaces(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
+	logger.Debug("Clearing openstack routers interfaces")
+	defer logger.Debugf("Exiting clearing openstack routers interfaces")
+
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return false, err
+	}
+
+	allRouters, err := getRouters(opts, filter, logger)
 	if err != nil {
 		logger.Error(err)
 		return false, nil
 	}
 
-	allRouters, err := routers.ExtractRouters(allPages)
+	numberToClear := len(allRouters)
+	numberCleared := 0
+	for _, router := range allRouters {
+		removed, err := removeRouterInterfaces(conn, filter, router, logger)
+		if err != nil {
+			logger.Debug(err)
+			continue
+		}
+		if !removed {
+			continue
+		}
```

Comment:
@mandre: This was the problem, function's result was ignored. This piece seems ugly, I'm happy to hear suggestions how to code it better.